src/Core/Services/Implementations/CipherService.cs

```diff
@@ -431,6 +433,13 @@ public async Task DeleteFolderAsync(Folder folder)
                     throw new BadRequestException("Not enough storage available for this organization.");
                 }
 
+                if (lastKnownRevisionDate != null && originalCipher.RevisionDate != lastKnownRevisionDate)
+                {
+                    throw new BadRequestException(
+                        "The cipher you are updating is out of date. Please save your work, sync your vault, and try again."
+                    );
+                }
+
```

Comment:
Can you use your existing ValidateCipherLastKnownRevisionDateAsync helper here?

Reply:
I could, but it would represent an extra read of the cipher since this call is unique in that a db read was already performed in the Controller layer

Reply:
I don't think you need to read from the database again. You should be able to always use the cipher that is passed in, which was read from the controller.

Reply:
@kspearrin you're 100% right. my thinking was that the ciphers given to these methods originate on the client, but for all `PUT` methods, they are database reads done on the controller level.